torch/distributed/tensor/_ops/_einsum_strategy.py

```diff
@@ -159,11 +169,6 @@ def gen_einsum_strategies(
     # generate strategies for entire mesh
     strategy_combs = itertools.product(*all_mesh_dim_strategies)
```

Comment:
@zpcore Can you clarify the design of this function? (It's more of a question about the pre-existing code than the changes you made).

1) the outer for loop over `mesh_dim in range(mesh.ndim):` appears to never use the local 'mesh_dim' inside its logic.  To me this suggests that we always generate the same 'mesh_dim_strategies' for each mesh dim.  
2) then we use itertols.product to create the combinations across mesh dims

it seems to me we could cut out the loop in (1) and then just use product directly, like `product((mesh_dim_strategies for _ in range(mesh_ndim))`